fix the order >=2 process of bigger files

vince62s opened the pull request:
related to issue https://github.com/kaldi-asr/kaldi/issues/837
this works for me with big audio file and one long utterance.

danpovey commented on egs/wsj/s5/steps/cleanup/make_utterance_graph.sh:
```diff
@@ -105,12 +105,13 @@ while read line; do
-      printf("\n"); }' > $wdir/text
-    ngram-count -text $wdir/text -order $ngram_order "$srilm_options" -lm - |\
+     echo $words > $wdir/text
+     split -b 200000 $wdir/text $wdir/xxx
```
I don't think this is very ideal because it may split in the middle of a word, leading to some words in the original text not being present in the LM that's produced.  How about this:
```
echo $words | perl -ane '@A = split; for ($n=0;$n<@A;$n++) { print "$A[$n] "; if(($n+1)%2 == 0 || $n+1==@A) {print "\n";} }' > $wdir/text

```

danpovey replied:
... also, I see that you have moved the dealing-with-OOVs into the ngram-count process.
This would be OK, except it's extremely inefficient to deal with the whole word-list on each utterance each time, if the utterances are short.  The original script had this problem too, as the wordlist was read by awk- I did not check it sufficiently when it was originally committed.  
The while loop is done <$text.  A much-more-efficient way to do this would be to pipe into the while loop, a command involving sym2int.pl (with the appropriate options to map unknown words) and then into int2sym.pl.

danpovey commented:
I'd still rather do it properly-- if it's an easy fix, why not do it?  There are edge cases that could cause a crash in your setup, e.g. splitting inside a UTF character; generating banned words or disambiguation symbols; etc.

I don't think you understood what I was saying, regarding the second, point.  I'm talking about the loop in the outer-level, enclosing while loop, that's done `<$text.`  I'm talking about fixing the OOV problem at that level, by piping the text through int2sym.pl and then sym2int.pl, with appropriate options, to get rid of the OOVs there.

danpovey commented:
Looks good, but can you please rebase your commits so there is just one commit?
I'm assuming you've tested this and it works.

danpovey commented:
... it's something like
git rebase -i HEAD~4
and change the 4 commits from 'pick pick pick pick' to something like "pick s s s" where s means squash.

vince62s commented:
Just rebased, but I think I need to push again ?
yes tested it and works. I had to change your perl line though because it adds a linefeed every 2 words which srilm does not like of course. So I changed to a reasonable cut-off value 30000 words.

danpovey commented:
Yes you may have to push; you will need the --force option as it's not a fast-forward push.
